src/groups/mqb/mqbplug/mqbplug_authenticator.h

```diff
@@ -0,0 +1,227 @@
+// Copyright 2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// mqbplug_authenticator.h                                            -*-C++-*-
+#ifndef INCLUDED_MQBPLUG_AUTHENTICATOR
+#define INCLUDED_MQBPLUG_AUTHENTICATOR
+
+//@PURPOSE: Provide base classes for the 'Authenticator' plugin.
+//
+//@CLASSES:
+//  mqbplug::AuthenticationData: interface for accessing data used for
+//                               authentication.
+//  mqbplug::AuthenticationResult: the results of authentication.
+//  mqbplug::Authenticator: interface for a plugin of type 'e_AUTHENTICATOR'.
+//  mqbplug::AuthenticatorPluginFactory: base class for the 'Authenticator'
+//                                       plugin factory.
+//
+//@DESCRIPTION: This component provide definitions for classes
+// 'mqbplug::AuthenticationData', 'mqbplug::AuthenticationResult',
+// 'mqbplug::Authenticator', 'mqbplug::AuthenticatorPluginFactory' used
+// as base classes for plugins that publish statistics and for their factories.
+
+// MQB
+#include <mqbcfg_brokerconfig.h>
+#include <mqbcfg_messages.h>
+#include <mqbplug_pluginfactory.h>
+
+// BDE
+#include <bdlbb_blob.h>
+#include <bsl_iostream.h>
+#include <bsl_string.h>
+#include <bslma_managedptr.h>
+#include <bslma_usesbslmaallocator.h>
+#include <bslmf_nestedtraitdeclaration.h>
+#include <bslstl_stringref.h>
+
+namespace BloombergLP {
+namespace mqbplug {
+
+// ========================
+// class AuthenticationData
+// ========================
+
+/// Interface for accessing data used during authentication in the broker.
+class AuthenticationData {
+  public:
+    // CREATORS
+
+    /// Destroy this object.
+    virtual ~AuthenticationData();
+
+    // ACESSORS
+
+    /// Return the authentication material provided by the client.
+    virtual const bsl::vector<char>& authPayload() const = 0;
+
+    /// Return the IP Address of the client.
+    virtual bslstl::StringRef clientIpAddress() const = 0;
+};
+
+// ==========================
+// class AuthenticationResult
+// ==========================
+
+/// The results of authentication.
+class AuthenticationResult BSLS_KEYWORD_FINAL {
```

Comment:
For this struct, I don't see why it should not be universal for all plugin types. If this is the case, we can keep one final implementation for it here.